src/Model.jl

```diff
@@ -94,28 +100,46 @@ function Model(lattice::AbstractMatrix{T},
                atoms::Vector{<:Element}=Element[],
                positions::Vector{<:AbstractVector}=Vec3{T}[];
                model_name="custom",
-               n_electrons::Int=sum(n_elec_valence, atoms; init=0),
+               n_electrons::Union{Int, Nothing}=nothing,
+               εF=nothing,
+               check_electrostatics=any(!iszero, charge_ionic.(atoms)),
                magnetic_moments=T[],
                terms=[Kinetic()],
                temperature=zero(T),
-               smearing=nothing,
+               smearing=default_smearing(temperature),
                spin_polarization=default_spin_polarization(magnetic_moments),
                symmetries=default_symmetries(lattice, atoms, positions, magnetic_moments,
                                              spin_polarization, terms),
                ) where {T <: Real}
-    lattice = Mat3{T}(lattice)
-    temperature = T(austrip(temperature))
+    # Ensembles and electrons
+    if isnothing(n_electrons) && isnothing(εF)  # Default: NVT with electrons given by atoms
+        n_electrons = default_n_electrons(atoms)
+    end
 
-    # Atoms and electrons
```

Comment:
put that whole block in the `else` clause of the previous thing, so it's clear it's related